types/node/globals.d.ts

```diff
@@ -239,6 +239,12 @@ declare class Buffer extends Uint8Array {
      */
     static from(data: number[]): Buffer;
     static from(data: Uint8Array): Buffer;
+    /**
+     * Creates a new buffer containing the coerced value of an object
+     * A `TypeError` will be thrown if {obj} has not mentioned methods or is not of other type appropriate for `Buffer.from()` variants.
+     * @param obj An object supporting `Symbol.toPrimitive` or `valueOf()`.
+     */
+    static from(obj: object, byteOffset?: number, length?: number): Buffer;
```

Comment:
Can we narrow this to something more specific?
As describe by the `@param` directive we could create an interface that supports either toPrimitive or valueOf?

Reply:
:+1:
Are you thinking something like: `{ valueOf(): any } | { [Symbol.toPrimitive](hint: string): any }` ?

Reply:
See https://github.com/nodejs/node/blob/master/lib/buffer.js#L311

I find the broadness of this function somewhat concerning, lol.

Reply:
I see,
`{ valueOf(): string | object } | { [Symbol.toPrimitive](hint: 'string'): string }`

lmao, yeah, its surprisingly vague